### PR TITLE
fix: Reading applicationState in the background

### PR DESCRIPTION
**Pull request by brustolin**
Keep track of application state by notification instead of reading directly from UIApplication. This way we avoid using UIApplication in the background thread when detecting an ANR.

_#skip-changelog_

### CODE REVIEW

File: Sources/Sentry/SentryUIApplication.m [hunk 1]
```diff
@@ -4,7 +4,34 @@
...
 
-@implementation SentryUIApplication
+@implementation SentryUIApplication {
+    UIApplicationState appState;
+}
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        [NSNotificationCenter.defaultCenter addObserver:self
```
Comment (philipphofmann) [inline Sources/Sentry/SentryUIApplication.m:14]: `m`: Please use the SentryNSNotificationCenterWrapper and add some tests when this solves the problem.

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 - Stop sending empty thread names (#3361)
 - Work around edge case with a thread info kernel call sometimes returning invalid data, leading to a crash (#3364)
+- Reading applicationState in the background (#3372)
```
Comment (philipphofmann) [inline CHANGELOG.md:13]: `m`: I don't this requires a changelog entry as it fixes a fix 😄

**Review by philipphofmann — APPROVED** [review]
LGTM.

File: CHANGELOG.md (hunk 2 above)
Comment (brustolin) [inline CHANGELOG.md:13]:
```suggestion
```
